Gem/Code/Source/Components/UI/UiMatchPlayerCoinCountsComponent.cpp

```diff
@@ -109,13 +109,23 @@ namespace MultiplayerSample
         {
             if (const PlayerIdentityComponent* identity = playerHandle.GetEntity()->FindComponent<PlayerIdentityComponent>())
             {
-                return identity->GetPlayerName();
+                PlayerNameString playerName = identity->GetPlayerName();
+                if (playerName.empty())
+                {
+                    if(const auto playerDebugId= playerHandle.GetEntity()->FindComponent<Multiplayer::NetworkDebugPlayerIdComponent>())
+                    {
+                        AZStd::string playerIdText = AZStd::string::format("<debug id: %i>", static_cast<uint32_t>(playerDebugId->GetPlayerId()));
+                        return playerIdText.c_str();
+                    }
+                    return "<player_identity_empty>";
+                }
+
+                return playerName;
             }
         }
 
-        return "Unknown Player";
+        return "<player_handle_doesn't_exist>";
```

Comment:
Nit: It's a bit weird to have the underscores but then use "doesn't". If there's something that can't handle spaces, it normally can't handle apostrophes either. Maybe "<player_handle_does_not_exist>"?

Reply:
Also, is this a style choice to make it look more roboty?

Reply:
Thanks! Yeah, <underscore> style choice just so we know something needs fixing and easy to find in code.

Reply:
fixed!